bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/speedtestHandler.java

```diff
@@ -0,0 +1,496 @@
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link speedtestHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Brian Homeyer - Initial contribution
+ */
+@NonNullByDefault
+public class speedtestHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(speedtestHandler.class);
+    private @Nullable speedtestConfiguration config;
+    private Gson GSON = new Gson();
+    private static Runtime rt = Runtime.getRuntime();
+    private long pollingInterval = 1440;
+    private String serverID = "";
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    public volatile boolean isRunning = false;
+
+    public static final String[] SHELL_WINDOWS = new String[] { "cmd" };
+    public static final String[] SHELL_NIX = new String[] { "sh", "bash", "zsh", "csh" };
+
+    private String speedTestCommand = "";
+    private static OS os = OS.NOT_SET;
+
+    private String ping_jitter = "";
+    private String ping_latency = "";
+    private String download_bandwidth = "";
+    private String download_bytes = "";
+    private String download_elapsed = "";
+    private String upload_bandwidth = "";
+    private String upload_bytes = "";
+    private String upload_elapsed = "";
+    private String isp = "";
+    private String interface_internalIp = "";
+    private String interface_externalIp = "";
+    private String result_url = "";
+    private String server_name = "";
+
+    /**
+     * Contains information about which operating system openHAB is running on.
+     */
+    public enum OS {
+        WINDOWS,
+        LINUX,
+        MAC,
+        SOLARIS,
+        UNKNOWN,
+        NOT_SET
+    }
+
+    public speedtestHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand channel: {} command: {}", channelUID, command);
+        String ch = channelUID.getId();
+        if (command instanceof RefreshType) {
+            updateChannels();
+            return;
+        }
+        if (ch.equals(speedtestBindingConstants.TRIGGER_TEST)) {
+            getSpeed();
+        }
+
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(speedtestConfiguration.class);
+        pollingInterval = config.refreshInterval;
+        serverID = config.serverID;
+        if (!config.execPath.isEmpty()) {
+            speedTestCommand = config.execPath;
+        } else {
+            switch (getOperatingSystemType()) {
+                case WINDOWS:
+                    speedTestCommand = "";
+                    break;
+                case LINUX:
+                case MAC:
+                case SOLARIS:
+                    speedTestCommand = "/usr/bin/speedtest";
```

Comment:
Should this be hardcoded? What happens for windows users as some people do run openHAB under windows. Can it be a config that defaults to this value perhaps?